nuget/lib/dependabot/nuget/file_parser/project_file_parser.rb

```diff
@@ -261,12 +266,53 @@ def build_dependency(name, req, version, prop_name, project_file, dev: false)
             requirement[:metadata] = { property_name: root_prop_name }
           end
 
-          Dependency.new(
+          dependency = Dependency.new(
             name: name,
             version: version,
             package_manager: "nuget",
             requirements: [requirement]
           )
+
+          # only include dependency if one of the sources has it
+          return unless dependency_has_search_results?(dependency)
```

Comment:
Consider logging the dependency and why it was skipped. It might be hard to track down why a certain dependency is missing without a log message here.